2.45 crashed on app_service_environment plan

Stack trace:
```
panic: runtime error: invalid memory address or nil pointer dereference
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x53abeb7]
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5:
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: goroutine 927 [running]:
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/web.flattenClusterSettings(0x0, 0x5eaa10f, 0x15)
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/web/app_service_environment_resource.go:493 +0x37
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/web.resourceAppServiceEnvironmentRead(0xc0004c0d20, 0x57682c0, 0xc0002c8900, 0x0, 0x0)
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/web/app_service_environment_resource.go:388 +0x86c
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc001281c20, 0xc00133a960, 0x57682c0, 0xc0002c8900, 0xc00000fbc0, 0x0, 0x0)
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:460 +0x129
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc0002085b8, 0x64ffdc0, 0xc001b8e8d0, 0xc00010c600, 0xc0002085b8, 0xc001b8e8d0, 0xc0026e2b78)
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:525 +0x3dd
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x5bd1340, 0xc0002085b8, 0x64ffdc0, 0xc001b8e8d0, 0xc00010c5a0, 0x0, 0x64ffdc0, 0xc001b8e8d0, 0xc000738e00, 0x37a)
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3269 +0x214
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00058b200, 0x6546ee0, 0xc00058b980, 0xc001875800, 0xc000d8fb90, 0x939f390, 0x0, 0x0, 0x0)
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1024 +0x522
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc00058b200, 0x6546ee0, 0xc00058b980, 0xc001875800, 0x0)
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1313 +0xd34
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00035a530, 0xc00058b200, 0x6546ee0, 0xc00058b980, 0xc001875800)
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:722 +0xa5
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-01-28T19:13:05.749+0100 [DEBUG] plugin.terraform-provider-azurerm_v2.45.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:720 +0xa5
```

--- a/azurerm/internal/services/web/app_service_environment_resource.go
+++ b/azurerm/internal/services/web/app_service_environment_resource.go
@@ -490,7 +490,7 @@ func expandAppServiceEnvironmentClusterSettings(input interface{}) *[]web.NameVa
 }
 
 func flattenClusterSettings(input *[]web.NameValuePair) interface{} {
-	if len(*input) == 0 {
+	if input == nil || len(*input) == 0 {
 		return []map[string]interface{}{}
 	}
 

